src/client.ts

```diff
@@ -53,21 +55,16 @@ export class APIClient {
   public async iterateUsers(
     iteratee: ResourceIteratee<OktaUser>,
   ): Promise<void> {
-    const users: OktaUser[] = [];
-    await this.oktaClient.listUsers().each((e) => {
-      users.push(e);
+    await this.oktaClient.listUsers().each(async (user) => {
+      await iteratee(user);
```

Comment:
Great, glad to see this can work! I just realized it could be even simpler.
```ts
await this.oktaClient.listUsers().each(iteratee);
```
The `iteratee` function has the interface expected by `each`.